olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp

```diff
@@ -51,13 +52,35 @@ std::string GetDatastoreServerUrl(const std::string& partition) {
              ? "https://api-lookup." + server_url->second + "/lookup/v1"
              : "";
 }
+
+class LookupApiLock {
+ public:
+  std::unique_lock<std::mutex> Lock(const client::HRN& catalog) {
+    std::unique_lock<std::mutex> lock(mutex_);
+    const std::string& key = catalog.ToString();
+    std::unique_lock<std::mutex> catalog_lock(mutexes_[key]);
+    return catalog_lock;
+  }
+
+ private:
+  std::mutex mutex_;
+  std::unordered_map<std::string, std::mutex> mutexes_;
+};
+
+// A temporary solution to avoid multiple online requests to lookup service for
+// the same catalog.
+static LookupApiLock gLookupLock;
```

Comment:
I am a little puzzled by the coding style here. If I read this correct here https://google.github.io/styleguide/cppguide.html#Constant_Names then also static variables should be names like constants, means `kLookupLock`. Right?